frontend/components/AgentSelection.tsx

```diff
@@ -25,29 +26,37 @@ type EachAgentProps = {
   agentConfig: AgentConfig;
 };
 
-const EachAgent = ({
-  showSelected,
-  agentType,
-  agentConfig,
-}: EachAgentProps) => {
-  const { goto: gotoSetup } = useSetup();
-  const { goto: gotoPage } = usePageState();
-  const { selectedAgentType, updateAgentType } = useServices();
-  const { masterSafes, isLoading } = useMasterWalletContext();
+const EachAgent = memo(
+  ({ showSelected, agentType, agentConfig }: EachAgentProps) => {
+    const { goto: gotoSetup } = useSetup();
+    const { goto: gotoPage } = usePageState();
+    const { selectedAgentType, updateAgentType } = useServices();
+    const { masterSafes, isLoading } = useMasterWalletContext();
 
-  const isCurrentAgent = showSelected ? selectedAgentType === agentType : false;
+    const isCurrentAgent = showSelected
+      ? selectedAgentType === agentType
+      : false;
 
-  const handleSelectAgent = useCallback(() => {
-    updateAgentType(agentType);
+    const handleSelectAgent = useCallback(async () => {
+      updateAgentType(agentType);
 
-    const isSafeCreated = masterSafes?.find(
-      (masterSafe) =>
-        masterSafe.evmChainId === AGENT_CONFIG[agentType].evmHomeChainId,
-    );
+      // DO NOTE REMOVE THIS DELAY
+      // NOTE: the delay is added so that agentType is updated in electron store
+      // and re-rendered with the updated agentType
+      await delayInSeconds(0.5);
```

Comment:
ahh interesting, love such hacks :D
but ideally would store a reference to the new value (or a state), update it inside the handleSelectAgent right after updating the store value, and then in a separate useEffect would check the value from store, compare it with the ref (or state) and call all the above

but this fine with me too, so approving

Reply:
it's not pointing to the same values, it's just the matter of synchronising them :D something similar to what we do on login when checking if the onboarding wasn't complete. just hope 0.5 sec delay is 100% stable